core/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -173,10 +173,17 @@ class Router(formatOps: FormatOps) {
         val skipSingleLineBlock =
           startsLambda || newlines > 0
 
+        val newlineBeforeLambdaParameters =
+          formatOps.style.newlineBeforeLambdaParameters
+
         Seq(
           Split(Space, 0, ignoreIf = skipSingleLineBlock)
             .withOptimalToken(close, killOnFail = true)
             .withPolicy(SingleLineBlock(close)),
+          Split(nl, 0, ignoreIf = !newlineBeforeLambdaParameters)
```

Comment:
I don't think we need a new branch here, it is probably enough to do something similar to what @melrief does in https://github.com/olafurpg/scalafmt/pull/440/files#diff-ded7f78a4a0d490270763a26fa35f6b7R828

Basically, we want to ignore the Space on line 180 if `newlines > 0`.